### PR TITLE
feat: enable build-secrets

### PR DESCRIPTION
**Pull request by tigarmo**
- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -16,9 +16,9 @@ codespell==2.2.5
 colorama==0.4.6
 commonmark==0.9.1
 coverage==6.5.0
-git+https://github.com/canonical/craft-application.git@main#egg=craft_application
+git+https://github.com/canonical/craft-application.git@CRAFT-2152-secrets-managed#egg=craft_application
```
Comment (tigarmo) [inline requirements-dev.txt:19]: remember to update this

**Comment by tigarmo** [comment]
Target branch TBD

**Review by cmatsuoka — APPROVED** [review]
Nice!

File: tests/spread/general/build-secrets/task.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  rm myfile.txt 
+
+  # Pack the ROCK  
+  # Check that the secret is masked on the "terminal"
+  run_rockcraft pack -v 2>&1 >/dev/null | grep -F "Using secret password: *****"
+
+  # Check that the secret is masked in the logfile.
+  rockcraft_log_file=$(find /root/.local/state/rockcraft/log/ -name 'rockcraft*.log' | sort -n | tail -n1)
+  grep -F "Using secret password: *****" "$rockcraft_log_file"
+  grep -F "gpg --passphrase ***** --batch -d myfile.txt.gpg" "$rockcraft_log_file"
```
Comment (sergiusens) [inline tests/spread/general/build-secrets/task.yaml:24]: newline :laughing:

**Review by sergiusens — APPROVED** [review]

**Pull request merged**